// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////////

syntax = "proto3";

package google.crypto.tink;

import "tink/proto/tink.proto";

option java_package = "com.google.crypto.tink.proto";
option java_multiple_files = true;
option go_package = "github.com/tink-crypto/tink-go/v2/proto/prf_based_deriver_go_proto";

message PrfBasedDeriverParams {
  KeyTemplate derived_key_template = 1;
}

message PrfBasedDeriverKeyFormat {
  KeyTemplate prf_key_template = 1;
  PrfBasedDeriverParams params = 2;
}

// key_type: type.googleapis.com/google.crypto.tink.PrfBasedDeriverKey
message PrfBasedDeriverKey {
  uint32 version = 1;
  KeyData prf_key = 2;
  PrfBasedDeriverParams params = 3;
}
